### PR TITLE
Add missing deprecation rerouting for `add_to_queue` in `TPUSpawnPlugin`

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Found while working on  #10034, #10059. 
This was missed in #9118

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda @kaushikb11 @tchaton @justusschock @awaelchli

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 1]
```diff
@@ -229,7 +229,12 @@ def __transfer_distrib_spawn_state_on_fit_end(self, trainer: "pl.Trainer", resul
                 self.mp_queue.put(best_model_path)
                 self.mp_queue.put(last_path)
                 self.mp_queue.put(results)
-                self.lightning_module.add_to_queue(self.mp_queue)  # adds the `callback_metrics` to the queue
+                # adds the `callback_metrics` to the queue
+                # TODO: Remove the if in v1.7
+                if is_overridden("add_to_queue", self.lightning_module):
```
Comment (tchaton) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:234]: Let's add an example for the `add_to_queue` example in the documentation. Ideally, Optuna with best metrics.

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:234]: in this PR? 

also related: #10865

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:234]: noted your request in the linked issue

**Review by justusschock — APPROVED** [review]

**Pull request merged**